packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,359 @@
+    // @dev The mapping is `depositId-->disputer-->Deposit`
+    mapping(uint256 => mapping(address => Deposit)) public disputedDeposits;
+
+    event DepositRelayed(
+        address indexed sender,
+        uint256 indexed depositTimestamp,
+        address recipient,
+        address indexed l1Token,
+        address slowRelayer,
+        uint256 amount,
+        uint256 proposerRewardPct,
+        uint256 realizedFeePct,
+        address depositContract
+    );
+    event RelaySpedUp(uint256 indexed depositId, address indexed fastRelayer, address indexed slowRelayer);
+    event FinalizedRelay(uint256 indexed depositId, address indexed caller);
+    event RelayDisputeSettled(uint256 indexed depositId, address indexed caller, bool disputeSuccessful);
     event ProvidedLiquidity(address indexed token, uint256 amount, uint256 lpTokensMinted, address liquidityProvider);
+
+    constructor(address _bridgePoolFactory, address _timer) Testable(_timer) {
+        bridgePoolFactory = BridgePoolFactory(_bridgePoolFactory);
+        require(bridgePoolFactory.finder() != address(0), "Invalid bridge pool factory");
+    }
+
+    /*************************************************
+     *          LIQUIDITY PROVIDER FUNCTIONS         *
+     *************************************************/
+
+    function deposit(address l1Token, uint256 amount) public {}
+
+    function withdraw(address lpToken, uint256 amount) public {}
+
+    /**************************************
+     *          RELAYER FUNCTIONS         *
+     **************************************/
+
+    /**
+     * @notice Called by Relayer to execute Slow relay from L2 to L1, fulfilling a corresponding deposit order.
+     * @dev There can only be one pending Slow relay for a deposit ID.
+     * @dev Caller must have approved this contract to spend the total bond for `l1Token`.
+     * @param depositId Unique ID corresponding to deposit order that caller wants to relay.
+     * @param depositTimestamp Timestamp of Deposit emitted by L2 contract when order was initiated.
+     * @param recipient Address on this network who should receive the relayed deposit.
+     * @param l1Token Token currency to pay recipient. This contract stores a mapping of `l1Token` to the
+     *     canonical token currency on the L2 network that was deposited to the Deposit contract.
+     * @param amount Deposited amount.
+     * @param realizedFeePct Computed offchain by caller, considering the amount of available liquidity for the token
+     * currency needed to pay the recipient and the count of pending withdrawals at the `depositTimestamp`. This fee
+     * will be subtracted from the `amount`. If this value is computed incorrectly, then the relay can be disputed.
+     * @param maxFeePct Maximum fee that L2 Depositor can pay. `realizedFee` <= `maxFee`.
+     * @param proposerRewardPct Reward % of deposit amount to pay relayers.
+     */
+    function relayDeposit(
+        uint256 depositId,
```

Comment:
We might also be able to drop this to a lower size depending on how it's created.